src/components/Pressable.tsx

```diff
@@ -0,0 +1,277 @@
+  (touch?.y ?? 0) > -(inset?.top ?? 0);
+
+const adaptStateChangeEvent = (
+  event: GestureStateChangeEvent<
+    HoverGestureHandlerEventPayload | LongPressGestureHandlerEventPayload
+  >
+): PressableEvent => {
+  const timestamp = Date.now();
+
+  const touchData = changeToTouchData(event);
+
+  const pressEvent = touchToPressEvent(touchData, timestamp);
+
+  return {
+    nativeEvent: {
+      touches: [pressEvent],
+      changedTouches: [pressEvent],
+      identifier: event.handlerTag,
+      locationX: event.x ?? -1,
+      locationY: event.y ?? -1,
+      pageX: event.absoluteX ?? -1,
+      pageY: event.absoluteY ?? -1,
+      target: 0, // node ID
+      timestamp: timestamp,
+      force: undefined,
+    },
+  };
+};
+
+const adaptTouchEvent = (event: GestureTouchEvent): PressableEvent => {
+  const timestamp = Date.now();
+
+  const nativeTouches = event.allTouches.map((touch) =>
+    touchToPressEvent(touch, timestamp)
+  );
+  const nativeChangedTouches = event.changedTouches.map((touch) =>
+    touchToPressEvent(touch, timestamp)
+  );
+
+  return {
+    nativeEvent: {
+      touches: nativeTouches,
+      changedTouches: nativeChangedTouches,
+      identifier: event.handlerTag,
+      locationX: event.allTouches.at(0)?.x ?? -1,
+      locationY: event.allTouches.at(0)?.y ?? -1,
+      pageX: event.allTouches.at(0)?.absoluteX ?? -1,
+      pageY: event.allTouches.at(0)?.absoluteY ?? -1,
+      target: 0, // node ID
+      timestamp: timestamp,
+      force: undefined,
+    },
+  };
+};
+
+export default function Pressable(props: PressableProps) {
+  const pressableRef = useRef<View>(null);
+  const [styleProp, setStyleProp] = useState<
+    StyleProp<ViewStyle> | undefined
+  >();
+  const [childrenProp, setChildrenProp] = useState<
+    React.ReactNode | undefined
+  >();
+
+  // disabled when onLongPress has been called
+  const isPressCallbackEnabled = useRef<boolean>(true);
+  const isPressedDown = useRef<boolean>(false);
+
+  const normalizedHitSlop: Insets =
+    typeof props.hitSlop === 'number'
+      ? numberAsInset(props.hitSlop)
+      : props.hitSlop ?? {};
+
+  const normalizedPressRetentionOffset: Insets =
+    typeof props.pressRetentionOffset === 'number'
+      ? numberAsInset(props.pressRetentionOffset)
+      : props.pressRetentionOffset ?? {};
+
+  const pressGesture = Gesture.LongPress().onStart((event) => {
+    if (isPressedDown.current) {
+      props.onLongPress?.(adaptStateChangeEvent(event));
+      isPressCallbackEnabled.current = false;
+    }
+  });
+
+  const setPressedState = (isPressed: boolean) => {
+    isPressedDown.current = isPressed;
+    if (typeof props.style === 'function') {
+      setStyleProp(props.style({ pressed: isPressed }));
+    } else {
+      setStyleProp(props.style);
+    }
+    if (typeof props.children === 'function') {
+      setChildrenProp(props.children({ pressed: isPressed }));
+    } else {
+      setChildrenProp(props.children);
+    }
+  };
```

Comment:
I don't think we should be keeping styles and children in the state. IMO it would be better to keep `pressedDown` as state, and move function checks (and execution) to the render function, [like in RN](https://github.com/facebook/react-native/blob/f7fe68861a52a0428698a9352d39a6e39afefe42/packages/react-native/Libraries/Components/Pressable/Pressable.js#L350-L352).

Reply:
Let me know if the current code is what you had in mind.

Reply:
Yes, just left one more comment regarding it but otherwise it's pretty good.